modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/EventSubscriber/WebtoolsAnalyticsEventSubscriber.php

```diff
@@ -96,16 +126,34 @@ public function analyticsEventHandler(AnalyticsEventInterface $event): void {
     $rules = $storage->loadMultiple();
     /** @var \Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRuleInterface $rule */
     foreach ($rules as $rule) {
-      if (preg_match($rule->getRegex(), $current_uri, $matches) === 1) {
-        $event->setSiteSection($rule->getSection());
-        $this->cache->set($current_uri, ['section' => $rule->getSection()], Cache::PERMANENT, $rule->getCacheTags());
-        // Currently there is no defined behavior for overlapping rules so we
-        // only take into account the first rule that applies.
-        return;
+      if ($rule->isSupportMultilingualAliases()) {
```

Comment:
If the functionality is confirmed to be applied to the current default language alias, we can just fetch it and apply the regexp on php level directly on it. At the moment this regexp will apply to all languages, so we could retrieve by mistake an alias meant for another page.

Reply:
Not all languages, all existing aliases for the source route. I was wrong.